Source/core/CyclicBuffer.cpp

```diff
@@ -265,7 +277,7 @@ namespace Core {
         AdminUnlock();
     }
 
-    uint32_t CyclicBuffer::Read(uint8_t buffer[], const uint32_t length)
+    uint32_t CyclicBuffer::Read(uint8_t buffer[], const uint32_t length, bool partialRead = false)
```

Comment:
I think this default parameter (for partialRead) should go inito the CyclicBuffer.h as otherwise the interface did change :-)